### ISSUE TITLE
[RRFC] `workspace-tag-version-prefix` config

### ISSUE THREAD

**Issue by ljharb** [npm/rfcs#570]
## Motivation ("The Why")
See https://github.com/npm/cli/issues/4017#issuecomment-1101481619 - `npm version` should be able to work inside workspaces.

### Example
See the OP in https://github.com/npm/cli/issues/4017.

I think adding an npm config of `workspace-tag-version-prefix`, which defaults to `[$workspaceName] ` - which would result in a tag of `[foo] v1.2.3`, for example - should address the need.

**Comment by ben-page** [comment]
`npm versions patch --ws` result in multiple commits or a single commit with multiple tags?

**Comment by ljharb** [comment]
I'd expect a single commit with multiple tags.

**Comment by ruyadorno** [comment]
in case we use a template system, it should probably also contain a value for version, e.g: `[$workspaceName] $workspaceVersion` in your original example @ljharb

**Comment by wraithgar** [comment]
this is a prefix so it's assumed version is the end so `[$workspaceName] v`

**Comment by ruyadorno** [comment]
good point @wraithgar I believe you're right, in case it's a prefix, yeah it should end at the **v**

**Comment by wraithgar** [comment]
> `npm version patch --ws` result in multiple commits or a single commit with multiple tags?

Multiple commits

**Comment by Kagu-chan** [comment]
> I'd expect a single commit with multiple tags.

I would like to see that as a default behavior which can be overwritten with a specific flag (like `git-single-tag` or something along the line).

Possible usecase:

In case of our company project we have a monorepo which deploys an API and releases a npm package (for connecting to this api and several more) at the same time, because they will be no cases where only one of both should be updated - we bundle it in one pipeline with tests and everything related triggered by tags.
Having multiple pipelines (because multiple tags will be generated) would result in longer pipelines due to dependecy testing.

**Comment by travi** [comment]
in case additional prior art is useful, [semantic-release-monorepo](https://github.com/pmowrer/semantic-release-monorepo) uses a [pattern of `<package-name>-<version>`](https://github.com/pmowrer/semantic-release-monorepo/#how) by default, but [enables adjusting the format](https://github.com/pmowrer/semantic-release-monorepo/#tagformat). 

example: `@scope/package-name-v9.25.0`
